backend/compact-connect/tests/smoke/query_provider_smoke_tests.py

```diff
@@ -0,0 +1,209 @@
+
+    get_provider_response = requests.get(
+        url=config.api_base_url + f'/v1/compacts/{compact}/providers/{provider_id}',
+        headers=staff_users_headers,
+        timeout=10,
+    )
+
+    if get_provider_response.status_code != 200:
+        raise SmokeTestFailureException(f'Failed to query provider. Response: {get_provider_response.json()}')
+    logger.info('Received success response from GET endpoint')
+
+    # Step 3: Verify the Provider response matches the profile.
+    provider_object = get_provider_response.json()
+
+    # verify the ssn is NOT in the response
+    if 'ssn' in provider_object:
+        raise SmokeTestFailureException(f'unexpected ssn field returned. Response: {get_provider_response.json()}')
+
+    # remove the fields from the user profile that are not in the query response
+    test_user_profile.pop('ssn', None)
+    test_user_profile.pop('dateOfBirth', None)
+    for provider_license in test_user_profile['licenses']:
+        provider_license.pop('ssn', None)
+        provider_license.pop('dateOfBirth', None)
+    for military_affiliation in test_user_profile['militaryAffiliations']:
+        military_affiliation.pop('documentKeys', None)
+
+    if provider_object != test_user_profile:
+        raise SmokeTestFailureException(
+            f'Provider object does not match the profile.\n'
+            f'Profile response: {json.dumps(test_user_profile)}\n'
+            f'Get Provider response: {json.dumps(provider_object)}'
+        )
+    logger.info('Successfully fetched expected provider records.')
+
+
+def query_provider_user_smoke_test():
+    """
+    Verifies that a provider record can be queried .
+
+    Step 1: Get the provider id of the provider user profile information.
+    Step 2: Have the staff user query for that provider using the profile information.
+    Step 3: Verify the Provider response matches the profile.
+    """
+
+    # Step 1: Get the provider id of the provider user profile information.
+    test_user_profile = call_provider_users_me_endpoint()
+    provider_id = provider_user_profile['providerId']
+    compact = provider_user_profile['compact']
+
+    # Step 2: Have the staff user query for that provider using the profile information.
+    staff_users_headers = get_staff_user_auth_headers(TEST_STAFF_USER_EMAIL)
+    post_body = {'query': {'providerId': provider_id}}
+
+    post_response = requests.post(
+        url=config.api_base_url + f'/v1/compacts/{compact}/providers/query',
+        headers=staff_users_headers,
+        json=post_body,
+        timeout=10,
+    )
+
+    if post_response.status_code != 200:
+        raise SmokeTestFailureException(f'Failed to query provider. Response: {post_response.json()}')
+    logger.info('Received success response from query endpoint')
+    # Step 3: Verify the Provider response matches the profile.
+    provider_object = post_response.json()['providers'][0]
```

Comment:
Perhaps worth adding a quick check to make sure at least (or exactly?) one provider is returned? (I realize this should never happen, but if/when a weird regression happens, a nice error message might help.)

```suggestion
    providers = post_response.json()['providers']
    if not providers:
        raise SmokeTestFailureException(f'No providers returned by query. Response: {post_response.json()}')
        
    provider_object = providers[0]
```